Node.js 22 issue: mbus-serial.c(90,23): warning C4090: 'function': different 'const' qualifiers

**Describe the bug**  

In node-wot we tried to add Node.js 22 as CI dependency and the run fails with
```
npm error   mbus-serial.c
npm error D:\a\node-wot\node-wot\node_modules\node-mbus\libmbus\mbus\mbus-serial.c(90,23): warning C4090: 'function': different 'const' qualifiers [D:\a\node-wot\node-wot\node_modules\node-mbus\build\libmbus.vcxproj]
```

**To Reproduce**  
see https://github.com/eclipse-thingweb/node-wot/pull/1274
https://github.com/eclipse-thingweb/node-wot/actions/runs/8878604576/job/24558179052?pr=1274#step:5:24

**Expected behavior**  
`npm ci` or `npm install` should work.

**Screenshots & Logfiles**  
https://github.com/eclipse-thingweb/node-wot/actions/runs/8878604576/job/24558179052?pr=1274#step:5:24

**Versions:**  
Tested node-mbus 2.2.2.

The file `/libmbus/mbus/mbus-serial.c` mentioned in the error log seems to be referenced here:
https://github.com/Apollon77/node-mbus/blob/17a3b5088aceb2280911fc60a7e36e3d20c79d29/binding.gyp#L42

--- a/.github/workflows/test-and-release.yml
+++ b/.github/workflows/test-and-release.yml
@@ -63,7 +63,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        node-version: [16.x, 18.x, 20.x]
+        node-version: [16.x, 18.x, 20.x, 22.x]
         os: [ubuntu-latest, windows-2019, windows-latest, macos-latest]
 
     steps:

--- a/.github/workflows/test-and-release.yml
+++ b/.github/workflows/test-and-release.yml
@@ -63,7 +63,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        node-version: [16.x, 18.x, 20.x, 22.x]
+        node-version: [16.x, 18.x, 20.x]
         os: [ubuntu-latest, windows-2019, windows-latest, macos-latest]
 
     steps:

--- a/binding.gyp
+++ b/binding.gyp
@@ -20,7 +20,13 @@
                         'OTHER_CFLAGS': [
                             '-w'
                         ],
-                    }
+                    },
+                    'defines': [
+                        # Stop <windows.h> from defining macros that conflict with
+                        # std::min() and std::max().  We don't use <windows.h> (much)
+                        # but we still inherit it from uv.h.
+                        'NOMINMAX',
+                    ]
                 }]
             ],
             'dependencies': [

--- a/binding.gyp
+++ b/binding.gyp
@@ -20,8 +20,11 @@
                         'OTHER_CFLAGS': [
                             '-w'
                         ],
-                    },
-                    'defines': [
+                    }
+
+                }],
+                ['OS=="win"', {
+			        'defines': [
                         # Stop <windows.h> from defining macros that conflict with
                         # std::min() and std::max().  We don't use <windows.h> (much)
                         # but we still inherit it from uv.h.

--- a/src/mbus-master.cc
+++ b/src/mbus-master.cc
@@ -255,7 +255,6 @@ public:
         char error[100];
         int address;
         int secondary_selected = 0;
-        int request_frame_res;
 
         memset((void *)&reply, 0, sizeof(mbus_frame));
 
@@ -376,7 +375,7 @@ public:
             Nan::New<String>(data).ToLocalChecked()
         };
         free(data);
-        callback->Call(2, argv);
+        callback->Call(2, argv, async_resource);
     };
 
     void HandleErrorCallback () {
@@ -388,7 +387,7 @@ public:
             Nan::Error(ErrorMessage())
         };
 
-        callback->Call(1, argv);
+        callback->Call(1, argv, async_resource);
     }
 private:
     char *data;
@@ -586,7 +585,7 @@ public:
             Nan::New<String>(data).ToLocalChecked()
         };
         free(data);
-        callback->Call(2, argv);
+        callback->Call(2, argv, async_resource);
     };
 
     void HandleErrorCallback () {
@@ -597,7 +596,7 @@ public:
         Local<Value> argv[] = {
             Nan::Error(ErrorMessage())
         };
-        callback->Call(1, argv);
+        callback->Call(1, argv, async_resource);
     }
 private:
     char *data;
@@ -769,7 +768,7 @@ public:
         Local<Value> argv[] = {
             Nan::Null()
         };
-        callback->Call(1, argv);
+        callback->Call(1, argv, async_resource);
     };
 
     void HandleErrorCallback () {
@@ -781,7 +780,7 @@ public:
             Nan::Error(ErrorMessage())
         };
 
-        callback->Call(1, argv);
+        callback->Call(1, argv, async_resource);
     }
 private:
     char *old_addr_str;

